### PR TITLE
Module migration

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Add v1alpha2 CRD as storage version
- Rename istio-sample CR to default
- Add migration script and documentation
- Fix possible flakiness of pull-istio-integration, because it wasn't using its own created operator image, but the image from pull-istio-operator-build(e.g. in [this job execution](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_istio/176/pull-istio-integration/1668922712514367488))

**Related issue(s)**
- https://github.com/kyma-project/istio/issues/168

### CODE REVIEW

**Review by videlov — CHANGES_REQUESTED** [review]

File: api/v1alpha2/istio_types.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// Valid IstioCR States.
+const (
+	Ready      State = "Ready"
+	Processing State = "Processing"
+	Error      State = "Error"
+	Deleting   State = "Deleting"
+)
+
+// Defines the desired specification for installing or updating Istio.
+type IstioSpec struct {
```
[the remark this answers is not in the document]
Comment (triffer) [reply api/v1alpha2/istio_types.go:34]: The Istio CRD includes version v1alpha1 from Kyma repo, so we are fine.

**Pull request merged**